homeassistant/components/apsystems/switch.py

```diff
@@ -0,0 +1,58 @@
+"""The power switch which can be toggled via the APsystems local API integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import DiscoveryInfoType
+
+from . import ApSystemsConfigEntry, ApSystemsData
+from .entity import ApSystemsEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ApSystemsConfigEntry,
+    add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the switch platform."""
+
+    add_entities([ApSystemsPowerSwitch(config_entry.runtime_data)])
+
+
+class ApSystemsPowerSwitch(ApSystemsEntity, SwitchEntity):
+    """The switch class for APSystems switches."""
+
+    _attr_device_class = SwitchDeviceClass.SWITCH
+
+    def __init__(
+        self,
+        data: ApSystemsData,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(data)
+        self._api = data.coordinator.api
+        self._attr_unique_id = f"{data.device_id}_power_switch"
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return super().available and self._api.get_device_power_status() is not None
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return state of the switch."""
+        return self._api.get_device_power_status()
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the switch on."""
+        await self.hass.async_add_executor_job(self._api.set_device_power_status(0))
+        await self._api.async_refresh()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the switch off."""
+        await self.hass.async_add_executor_job(self._api.set_device_power_status(1))
+        await self._api.async_refresh()
```

Comment:
Not sure why we call async_refresh

Reply:
The API doesn't have an `async_refresh` function.

Reply:
Lol

Reply:
Oops, sorry. All this async stuff is still partly "magic" for me and it's not always clear to me when a function implies a certain functionality just by its function name and when what is executed where.

Basically I just wanted to update the coordinator to get the current switch state after switching the switch.
However, after thinking about it again, I think I don't need to do this at all, as the update process is handled automatically via the `is_on` function.

If I annoy you with my trial and error commits, feel free to ignore the merge request for now. I will remove it from the draft status later.
If it still causes too much noise, I will gladly close it again to reopen it later.

Best regards and thanks for your feedback. 😊